groups.tf

```diff
@@ -0,0 +1,48 @@
+resource "aws_security_group" "this" {
+  name_prefix = var.name
+  description = "Attached to all Bitbucket instances"
+  vpc_id      = var.vpc_id
+
+  tags = merge(
+    var.tags,
+    map(
+      "Name", "${var.name}"
+    )
+  )
+
+  lifecycle {
+    create_before_destroy = true
+  }
+}
+
+resource "aws_security_group_rule" "allow_all" {
+  count             = var.asg_allow_outbound_egress ? 1 : 0
+  cidr_blocks       = ["0.0.0.0/0"] #tfsec:ignore:AWS007
+  description       = "Allow outbound egress"
+  from_port         = 0
+  protocol          = "-1"
+  security_group_id = aws_security_group.this.id
+  to_port           = 0
+  type              = "egress"
+}
```

Comment:
I made it conditional so that you could turn it off, since this is a module. Bitbucket has several outbound ports it likes to talk on. A user could optionally turn this off and supply their own egress group that is more granular, so I think this is supportive of that.